src/components/FeatureCards/FeatureCard.tsx

```diff
@@ -119,13 +120,34 @@ export const FeatureCard = ({ data, isSuccess }: FeatureCardProps) => {
     });
   };
 
+  // Type guard for SpindlTrackData
+  const isSpindlTrackData = (data: unknown): data is SpindlTrackData => {
+    return (
+      typeof data === 'object' &&
+      data !== null &&
+      'impression_id' in data &&
+      'ad_creative_id' in data
+    );
+  };
+
+  const handleSpindl = (spindlData: SpindlTrackData) => {
+    trackSpindl(spindlData.impression_id, spindlData.ad_creative_id);
+  };
+
   const handleCTA = (
     event: React.MouseEvent<HTMLAnchorElement, MouseEvent>,
   ) => {
     event.stopPropagation();
-    !data?.attributes?.DisplayConditions?.hasOwnProperty('showOnce') &&
-      !!data?.attributes?.uid &&
+
+    // Mark feature card as disabled if needed
+    if (
+      !data?.attributes?.DisplayConditions?.hasOwnProperty('showOnce') &&
```

Comment:
much better to use what you've used above this, hasOwnProperty is very much vanilla js and very similar (let's keep consistency)